php/libraries/Permission.class.inc

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * This file contains the Permission class.
+ *
+ * PHP Version 5-7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   Rida Abou-Haidar <rida.loris@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+/**
+ * The Loris Permission class
+ *
+ * @category Main
+ * @package  Loris
+ * @author   Rida Abou-Haidar <rida.loris@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Permission
+{
+    /**
+     * Stores Database being used
+     *
+     * @var    $DB Database
+     * @access private
+     */
+    var $DB;
+
+    /**
+     * Permission constructor.
+     *
+     * @param Database $Database database
+     */
+    function __construct($Database)
+    {
+        $this->DB = $Database;
+    }
+
+
+    /**
+     * Gets the list of permissions in the database.
+     *
+     * @return array Associative array in the form $permissionID=>$permissionName
+     */
+    function getPermissions()
+    {
+        $permissions = $this->DB->pselectColWithIndexKey(
+            "SELECT permID, code
+         FROM permissions",
+            array(),
+            "permID"
+        );
+
+        return $permissions;
+    }
+
+    /**
+     * Gets the list of permission labels in the database.
+     *
+     * @return array Associative array in the form $permissionID=>$permissionLabel
+     */
+    function getPermissionLabels()
+    {
+        $permissions = $this->DB->pselectColWithIndexKey(
+            "SELECT permID, description
+         FROM permissions",
+            array(),
+            "permID"
+        );
+
+        return $permissions;
+    }
+
+    /**
+     * Checks if the string is a permission within the database
+     *
+     * @param string $permissionName the permission to be checked
+     *
+     * @return boolean
+     */
+    function isPermission($permissionName)
+    {
+        $permissions = $this->getPermissions();
+        if (in_array($permissionName, $permissions, true)) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
multiple return statements in a single function.

Reply:
only one is ever reached though... is that a problem ?

Reply:
Not a problem by itself, probably just me that is 'old school'. Some C compiler required only one return statement at the end of the function. Was considered bad practice to have multiple return statement in a single function.

We had to use a variable and return the variable value in these case.

Reply:
https://hackerchick.com/religious-war-48293-single-vs-multiple/

let the war begin.... :trollface:

Reply:
You could avoid the whole debate by just getting rid of the if statement entirely:

`return in_array($permissionName, $permissions, true);`

Reply:
@ridz, Where do we put blue helmet soldier here?
`           return true;
      }
     // UN troups
  return false;
}`